lib/linux_admin/registration_system.rb

```diff
@@ -36,7 +35,7 @@ def self.registration_type_uncached
 
     def self.white_list_methods
       @white_list_methods ||= begin
-        all_methods = RegistrationSystem.instance_methods(false) + Rhn.instance_methods(false) + SubscriptionManager.instance_methods(false)
+        all_methods = RegistrationSystem.instance_methods(false) + SubscriptionManager.instance_methods(false)
```

Comment:
Is there a reason you are keeping this this class in here at all?  The only value this provided was that there were two different `RegistrationSystem` classes in the code base prior where one could fallback to the other.  Now you just have a bunch of complexity for no reason.

Seems like you could ditch this class entirely, and move the `lib/linux_admin/registration_system/subscription_manager.rb` file up to the base folder.  Would allow us to get rid of all of this `method_missing` business.